### PR TITLE
Scaffold an official project structure on with `meteor create`

### PR DESCRIPTION
**Pull request by mhidou**
Hi,

This PR concerns the issue #6974 

As a full stack developer, I enjoy to start my projects with the smallest amount of code to avoid spending time deleting what I don't need. So I've choose to create a minimal scaffold with architecture based on meteor guide, to help rookies understanding good practices.

Now when you want create a new meteor app you can add the --bare option if you want an empty app or you can add the --full option to create a full app.

In the full version I added the `kadira:flow-router`, `kadira:blaze-layout` and the `less` packages.

You can check this [repo](https://github.com/mhidou/meteor-skel-full) to see the boilerplate.

Hope that you will enjoy what I've done, it's always a pleasure to help the Meteor project.
- **Basic behavior of the create command**

```
$ meteor create test
Created a new Meteor app in 'test'.

To run your new app:
  cd test
  meteor

If you are new to Meteor, try some of the learning resources here:
  https://www.meteor.com/learn

meteor create --bare to create an empty app.
meteor create --full to create a scaffolded app.

$ find test -type f
test/.meteor/local/...
test/.meteor/packages
test/.meteor/platforms
test/.meteor/release
test/.meteor/versions
test/client/main.css
test/client/main.html
test/client/main.js
test/package.json
test/server/main.js
```
- **The --bare option**

```
$ meteor/meteor create --bare test-bare
Created a new Meteor app in 'test-bare'.

To run your new app:
  cd test-bare
  meteor

If you are new to Meteor, try some of the learning resources here:
  https://www.meteor.com/learn

$ find test-bare -type f
test-bare/.meteor/local/...
test-bare/.meteor/packages
test-bare/.meteor/platforms
test-bare/.meteor/release
test-bare/.meteor/versions
test-bare/package.json
```
- **The --full option**

```
$ meteor/meteor create --full test-full
Created a new Meteor app in 'test-full'.

To run your new app:
  cd test-full
  meteor

If you are new to Meteor, try some of the learning resources here:
  https://www.meteor.com/learn

$ find test-full -type f
test-full/.meteor/local/...
test-full/.meteor/packages
test-full/.meteor/platforms
test-full/.meteor/release
test-full/.meteor/versions
test-full/client/head.html
test-full/client/main.js
test-full/client/main.less
test-full/imports/api/links/links.js
test-full/imports/api/links/links.tests.js
test-full/imports/api/links/methods.js
test-full/imports/api/links/methods.tests.js
test-full/imports/api/links/server/publications.js
test-full/imports/api/links/server/publications.tests.js
test-full/imports/startup/both/index.js
test-full/imports/startup/both/useraccounts-configuration.js
test-full/imports/startup/client/index.js
test-full/imports/startup/client/routes.js
test-full/imports/startup/server/fixtures.js
test-full/imports/startup/server/index.js
test-full/imports/startup/server/register-api.js
test-full/imports/ui/components/hello/hello.html
test-full/imports/ui/components/hello/hello.js
test-full/imports/ui/components/info/info.html
test-full/imports/ui/components/info/info.js
test-full/imports/ui/layouts/body/body.html
test-full/imports/ui/layouts/body/body.js
test-full/imports/ui/pages/home/home.html
test-full/imports/ui/pages/home/home.js
test-full/imports/ui/pages/not-found/not-found.html
test-full/imports/ui/pages/not-found/not-found.js
test-full/imports/ui/stylesheets/not-found.less
test-full/package.json
test-full/private/README.md
test-full/public/img/404.svg
test-full/public/img/bg-footer.svg
test-full/server/main.js
```
- **Man**

```
$ meteor help create
Usage: meteor create [--release <release>] [--bare|--full] <path>
       meteor create [--release <release>] --example <example_name> [<path>]
       meteor create --list
       meteor create --package [<package_name>]

Make a subdirectory named <path> if it doesn't exist and create a new Meteor app
there. You can pass an absolute path, relative path, or '.' for the current
directory. Use the --bare option to create an empty app. To scaffold the app
use the --full option.

With the --package option, creates a Meteor package instead of an app. If you're
in an app, the package will go in the app's top-level 'packages' directory;
otherwise it will be created in the current directory.

The app will use the release of Meteor specified with the --release
option, or the latest available version if the option is not specified. (A
package created in an app, will be created using the application's version of
meteor and a package created outside a meteor app will use the latest release).

You can pass --example to start off with a copy of one of the Meteor
sample applications. Use --list to see the available examples. There are
currently no package examples.

Options:
  --package  Create a new meteor package instead of an app.
  --example  Example template to use.
  --list     Show list of available examples.
  --bare     Create an empty app.
  --full     Create a scafolded app.
```

### CODE REVIEW

**Comment by tmeasday** [comment]
Hi @mhidou can you please rebase this PR against current devel? The commit you started off had some problems running the Circle tests.

**Comment by mhidou** [comment]
@tmeasday rebase done

File: tools/static-assets/skel-full/imports/startup/both/useraccounts-configuration.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+// The useraccounts package must be configured for both client and server to work properly.
```
Comment (tmeasday) [inline tools/static-assets/skel-full/imports/startup/both/useraccounts-configuration.js]: This app doesn't include this package, so this seems odd -- I think maybe you could just add a comment to the `imports/startup/both/index.js` detailing the kind of shared configuration you might have.

File: tools/static-assets/skel-full/imports/api/links/server/publications.tests.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Tests for the links publications
```
Comment (tmeasday) [inline tools/static-assets/skel-full/imports/api/links/server/publications.tests.js]: Shall we properly include tests?

File: tools/static-assets/skel-full/imports/api/links/server/publications.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// All link-related publications
+//
+// TODO: Remove the autopublish package before adding publications
```
Comment (tmeasday) [inline tools/static-assets/skel-full/imports/api/links/server/publications.js]: I think removing autopublish + insecure in this app is a great idea.

File: tools/static-assets/skel-full/imports/api/links/server/publications.tests.js (hunk 2 above)
Comment (tmeasday) [inline tools/static-assets/skel-full/imports/api/links/server/publications.tests.js]: (Like maybe we should include the publication collection package and the validated method stuff for method tests?)

File: tools/static-assets/skel-full/imports/api/links/server/publications.js (hunk 3 above)
Comment (mhidou) [inline tools/static-assets/skel-full/imports/api/links/server/publications.js]: I agree

**Comment by mhidou** [comment]
Pr updated

**Comment by stubailo** [comment]
> If you are new to Meteor, try some of the learning resources here:

I'd say the `--full` should probably link to some page in the Meteor Guide that describes all of the stuff it has created.

**Comment by tmeasday** [comment]
Sorry about the slow response here, this looks great now @mhidou. I'm sure there's a lot of bike-shedding that could happen here (which is why I am still a little tempted to have it in a separate repo somehow where that could happen more quickly), but I think this will help people a lot.

**Comment by tmeasday** [comment]
Thanks for this @mhidou! Merged!

**Comment by dburles** [comment]
Hmm, the code style could be improved. Should it not adhere to the airbnb styleguide as recommended by the Meteor guide itself?

**Comment by tmeasday** [comment]
Oh, please, send PRs to improve it!

**Comment by theodorDiaconu** [comment]
I think this is outside Meteor. Sorry but this is opinionated. I could easily create an advanced scaffolding tool in bash. Without having to affect Meteor code. I would rather see Meteor point to those libraries than build this inside it.

If you use Redux + React + Meteor you'll have a different file structure. If you build a trully large app, then again, different structure, maybe: imports/api, imports/ui => imports/submodule/api, imports/submodule/ui.

If you're building an app with a micro-service architecture... You get the idea, you may need to do it old-school with different PACKAGE_DIRS 

Depending on the UI layer, React/Blaze/Vue/Angular2 you'll have ofcourse a difference ui structure.

**Comment by mhidou** [comment]
@theodorDiaconu I think that the scaffolded app is for rookies, it will help them to start with [meteor guide files structure](https://guide.meteor.com/structure.html) (a kind of example app) of course if you are building a large app this scaffold can be useless.

A scaffolding tool like in rails will be awesome. It will helps a lot of users and saves tons of time.

#5283

**Comment by abernix** [comment]
This doesn't break the ability to do micro-service architecture and it's certainly not going to work for every app, but as @mhidou stated it is intended to match up relatively well with what the Meteor Guide suggests for an app structure.

I do think it would be nice to keep these outside of the Meteor core repository but in the spirit of the long-standing `meteor create` command, this does seem on par with what `meteor-tool` would be expected to do.

As the task of making these 'import-able' from another repo seems like a much larger task, and to keep this lean and see if it's even popular, it seems harmless to include these here for now.  I do agree the code style should be improved as @dburles pointed out though.

**Comment by dburles** [comment]
Hey guys so just looking into working on the code style a little. Just wondering how are the scaffold `package.json` dependencies going to be kept up to date with the Meteor release? can it be automated?

**Comment by tmeasday** [comment]
The `meteor-node-stubs` package @dburles? It doesn't change much, I'm sure we can change all three apps when we update it..

**Comment by dburles** [comment]
Yep and `babel-runtime` I guess now too.

**Pull request merged**